lib: nrf_modem: trace: add NRF_MODEM_LIB_TRACE_LEVEL_OFF

gregersrygg opened the pull request:
Add new kconfig setting to enable traces, but disabled by default. This allows trace functionality for the modem to be compiled in, but only activated when a specific condition is met during runtime.

NordicBuilder commented:
## Test specification

### [CI/Jenkins/NRF](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-10149/1/)
- Integration Platforms

### [CI/Jenkins/integration](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-10149/1/)
| Test Module | File based changes | Manually selected | West overwrite |
| ----------- | :----------------: | :---------------: | :------------: |
| test-fw-nrfconnect-nrf-iot_lwm2m | X | | |
| test-fw-nrfconnect-nrf-iot_mosh | X | | |
| test-fw-nrfconnect-nrf-iot_samples | X | | |
| test-fw-nrfconnect-nrf-iot_zephyr_lwm2m | X | | |


[Detailed information of selected test modules](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-10149/1/artifact/test-spec/test-spec-summary.yml/*view*/)

*Note: This message is automatically posted and updated by the CI*

eivindj-nordic reviewed (commented):
Kconfig ooks good. It would be nice to have an entry in the documentation as well, in `nrf_modem_lib.rst` with the other levels under `modem_trace_module`.

gregersrygg commented:
> Kconfig ooks good. It would be nice to have an entry in the documentation as well, in nrf_modem_lib.rst with the other levels under modem_trace_module.

Good point. I have updated the documentation page and added a note to the release notes.

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-10149
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

divipillai commented on doc/nrf/releases/release-notes-changelog.rst:
```diff
@@ -493,6 +493,10 @@ Modem libraries
 
+  * Added:
+
+    * :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE_LEVEL_OFF` to set modem trace level to off by default.
```
```suggestion
    * The :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE_LEVEL_OFF` Kconfig option to set the modem trace level to off by default.
```
